mobify/platform-scaffold
# Use knownModals to avoid mistaking the persistentModals key as a modal

Our current getModals returns all keys in the modals branch of the redux store. The problem is that a recent change added a persistentModals key to this branch is not a reference to a specific modal, but the scaffold treats it as if it is in fact a modal.

**JIRA**: [WEB-1562](https://mobify.atlassian.net/browse/WEB-1562)

**Linked PRs**: [progressive-web-sdk#876](https://github.com/mobify/progressive-web-sdk/pull/876)

## Changes
- Switch to the new `getKnownModals` selector to ensure that the recently added `persistentModals` key isn't confused as an open modal for the purpose of hiding/showing the modal background.

## Todos

## How to test-drive this PR
- Change to your cloned SDK repository (see linked PR above)
- Run `npm run prod:build`
- Run `npm link`
- Hop back to this repository
- Run `npm link progressive-web-sdk`
- Run `npm start`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- **Verify** that the `div + .pw-dangerous-html` element is set to `aria-hidden="false"`
- Open the main navigation
- **Verify** that the `div + .pw-dangerous-html` element is set to `aria-hidden="true"`
- **With a Screen Reader**
    - Refresh the site
    - **Verify** that the site can be "tabbed" to with VoiceOver (a.k.a. the site isn't completely hidden)
